extensions/civetweb/processors/ListenHTTP.cpp

```diff
@@ -27,11 +27,22 @@ namespace minifi {
 namespace processors {
 
 core::Property ListenHTTP::BasePath(
-    core::PropertyBuilder::createProperty("Base Path")->withDescription("Base path for incoming connections")->isRequired(false)->withDefaultValue<std::string>("contentListener")->build());
+    core::PropertyBuilder::createProperty("Base Path")->
+    withDescription("Base path for incoming connections")->
+    isRequired(false)->
+    withDefaultValue<std::string>("contentListener")->build());
 
 core::Property ListenHTTP::Port(
-    core::PropertyBuilder::createProperty("Listening Port")->withDescription("The Port to listen on for incoming connections")->isRequired(true)->withDefaultValue<int>(
-        80, core::StandardValidators::PORT_VALIDATOR())->build());
+    core::PropertyBuilder::createProperty("Listening Port")->
+    withDescription("The Port to listen on for incoming connections")->
+    isRequired(true)->
+    withDefaultValue<int>(80, core::StandardValidators::PORT_VALIDATOR())->build());
+
+core::Property ListenHTTP::RandomPort(
+    core::PropertyBuilder::createProperty("Random Port")->
+    withDescription("Choose random port to listen - for testing purposes only!")->
```

Comment:
I don't think we should have testing code available to production environments. This will be relayed through the C2 framework. Perhaps eliminating it when not building tests will work.

Reply:
Good point. Added an ifdef to only add it to the set of supported properties in case of building with tests.